Mathlib/FieldTheory/PrimitiveElement.lean

```diff
@@ -282,75 +273,49 @@ private lemma finite_intermediateField_of_exists_primitive_element.aux_2
     (heq : (minpoly K α).map (algebraMap K E) = (minpoly K' α).map (algebraMap K' E)) :
     K ≤ K' := by
   set g := (minpoly K α).map (algebraMap K E)
-  set K'' := K ⊓ K'
-  let p : Polynomial K'' := Finset.sum g.support fun n => C ⟨g.coeff n,
-    (mem_inf (S := K) (T := K')).2
-      ⟨by rw [coeff_map]; exact ((minpoly K α).coeff n).property,
-        by rw [heq, coeff_map]; exact ((minpoly K' α).coeff n).property⟩⟩ * X ^ n
-  have hp : p.map (algebraMap K'' E) = g := by
-    rw [← sum_C_mul_X_pow_eq g, Polynomial.sum_def, Polynomial.map_sum]
-    congr
-    funext n
-    rw [Polynomial.map_mul, Polynomial.map_pow, Polynomial.map_X, Polynomial.map_C]
-    congr
-  let q := minpoly K'' α
-  obtain ⟨r, hqdvdp⟩ := show q ∣ p by
-    apply minpoly.dvd
-    rw [aeval_def, eval₂_eq_eval_map, hp, ← eval₂_eq_eval_map, ← aeval_def]
+  have g_lifts : g ∈ lifts (algebraMap ↑(K ⊓ K') E)
```

Comment:
[Polynomial.lifts](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Data/Polynomial/Lifts.html#Polynomial.lifts) is exactly designed for such situation but is little known. Sorry for not mentioning earlier!